packages/@react-aria/overlays/src/useModal.tsx

```diff
@@ -106,16 +106,29 @@ export function OverlayProvider(props: ModalProviderProps) {
   );
 }
 
+interface OverlayContainerProps extends ModalProviderProps {
+  getContainerRootElement?: () => HTMLElement
```

Comment:
I'm curious - why a getter has been chosen for this? It's resolved in render anyway and since this is passed to the `createPortal` it looks a little bit odd as it's of a different type than what `createPortal` accepts.

Reply:
hmm... I thought there was an explicit reason for that, but I don't see a reference to it and your suggestion makes sense to me. I'll double check with the team to make sure I'm not forgetting something

Reply:
agreed. I think it's weird that it's a function.